lnbits/wallets/phoenixd.py

```diff
@@ -0,0 +1,169 @@
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("/getinfo", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return StatusResponse(error_message, 0)
+
+        data = int(r.json()["channels"][0]["balanceSat"]) * 1000
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
+
+        msats_amount = amount
+        data: Dict = {
+            "amountSat": f"{msats_amount}",
+            "description": memo,
+            "externalId": "",
+        }
+
+        r = await self.client.post(
+            "/createinvoice",
+            data=data,
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()
+        # logger.info(f'data: {data}')
+
+        checking_id = data["paymentHash"]
+        payment_request = data["serialized"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(
+        self, bolt11_invoice: str, fee_limit_msat: int
+    ) -> PaymentResponse:
+        r = await self.client.post(
+            "/payinvoice",
+            data={
+                "invoice": bolt11_invoice,
+            },
+            timeout=40,
+        )
+
+        if r.is_error:
+            logger.error(f"pay_invoice error: {r.json()}")
+            error_message = r.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+        logger.info(f"pay_invoice data: {data}")
+
+        checking_id = data["paymentHash"]
+        fee_msat = -int(data["routingFeeSat"])
+        preimage = data["paymentPreimage"]
+
+        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+
+    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+        r = await self.client.get(f"/payments/incoming/{checking_id}")
+        if r.is_error:
+            return PaymentPendingStatus()
+        data = r.json()
+        # logger.info(f'get_invoice_status data: {data}')
+
+        fee_msat = data["fees"]
+        preimage = data["preimage"]
+        is_paid = data["isPaid"]
+
+        return PaymentStatus(paid=is_paid, fee_msat=fee_msat, preimage=preimage)
```

Comment:
This is a good example of how `paid` should be set. Less room for error.

Reply:
ok